particles/Tutorial/Kotlin/Blackjack/Blackjack.arcs

```diff
@@ -0,0 +1,64 @@
+// Encodes a card from 0 to 51. -1 indicates an invalid (or joker).
+schema Card
+  value: Number
+
+// Encodes the cards in the deck as a string. The cards are assumed to be ordered and the
+// character at an index encodes whether the card is still present in the deck or not.
+schema CardDeck
+  cards: Text
+
+// Card request message.
+schema CardRequest
+  player: Text
+
+// Result of a card request from the player
+schema CardRequestResult
+  player: Text
+  card: Number
+
+// A dealing shoe in the game.
+particle DealingShoe in 'Blackjack.wasm'
+  decks: reads writes CardDeck
```

Comment:
Does decks need to be an entity? It looks like it is all internal to DealingShoe, so I think you could just store it locally within the Kotlin object.

Reply:
I don't think it has to be. I started with an entity thinking that it would be needed for persistence. e.g., the particle goes down and restarts from the last known state. Is that the right understanding of how entities would work? Please correct me if I am wrong. 

If we don't need an entity here, I guess we don't need an entity for the `hand` in the `Player` and `Dealer` particles as well?

Reply:
Ah yes. I wasn't thinking about persistence. I might change decks be `[card]` then, but I think this is a personal preference. It just fits better in my mental model of the game. This might also simplify dealing the cards.

Reply:
I will do this in a follow up PR?